lib/ansible/modules/cloud/cloudstack/cs_physical_network.py

```diff
@@ -0,0 +1,493 @@
+            if 'errortext' in res:
+                self.module.fail_json(msg="Failed: '%s'" % res['errortext'])
+            for loadbalancer in res['internalloadbalancerelement']:
+                self.loadbalancers[loadbalancer['nspid']] = loadbalancer
+
+            if nspid not in self.loadbalancers:
+                self.module.fail_json(msg="Failed: No Loadbalancer found for nsp '%s'" % nsp_name)
+
+        return self.loadbalancers[nspid]
+
+    def set_vrouter_element_state(self, enabled, nsp_name='VirtualRouter'):
+        vrouter = self.get_vrouter_element(nsp_name)
+        if vrouter['enabled'] == enabled:
+            return vrouter
+
+        args = {
+            'id': vrouter['id'],
+            'enabled': enabled
+        }
+        res = self.cs.configureVirtualRouterElement(**args)
+        if 'errortext' in res:
+            self.module.fail_json(msg="Failed: '%s'" % res['errortext'])
+
+        self.result['changed'] = True
+        return res
+
+    def set_loadbalancer_element_state(self, enabled, nsp_name='InternalLbVm'):
+        loadbalancer = self.get_loadbalancer_element(nsp_name=nsp_name)
+        if loadbalancer['enabled'] == enabled:
+            return loadbalancer
+
+        args = {
+            'id': loadbalancer['id'],
+            'enabled': enabled
+        }
+        res = self.cs.configureInternalLoadBalancerElement(**args)
+        if 'errortext' in res:
+            self.module.fail_json(msg="Failed: '%s'" % res['errortext'])
+
+        self.result['changed'] = True
+        return res
+
+    def present_network(self):
+        network = self.get_network()
+        if network:
+            network = self._update_network()
+        else:
+            network = self._create_network()
+        return network
+
+    def _create_network(self):
+        required_params = [
+            'name',
+            'zone',
+        ]
+        self.module.fail_on_missing_params(required_params=required_params)
+
+        args = self._get_common_args()
+        args['zoneid'] = self.get_zone(key='id')
+        args['domainid'] = self.get_domain(key='id')
+        args['url'] = self.module.params.get('url')
+        args['username'] = self.module.params.get('username')
+        args['password'] = self.module.params.get('password')
+
+        self.result['changed'] = True
+
+        id = None
+        if not self.module.check_mode:
+            res = self.cs.createPhysicalNetwork(**args)
+            if 'errortext' in res:
+                self.module.fail_json(msg="Failed: '%s'" % res['errortext'])
+            id = res['id']
+        return id
+
+    def _update_network(self):
+        network = self.get_network()
+
+        args = self._get_common_args()
+        args['id'] = network['id']
+
+        if self.has_changed(args, network):
+            self.result['changed'] = True
+
+            if not self.module.check_mode:
+                res = self.cs.updatePhysicalNetwork(**args)
```

Comment:
The `physcialNetwork` API is an async API, that is why the `poll_async` function should be used (e.g. https://github.com/ansible/ansible/blob/devel/lib/ansible/modules/cloud/cloudstack/cs_vpc.py#L334-L336) to query the updated resource and return it to the user.